### PR TITLE
fixes #543 Avoid logLock when determining WALs in use

### LINKED ISSUE #543: Tablet assignment thread gets stuck
Received this exception while running continuous ingest on Accumulo 1.9 with heavy tserver agitation.

```
Assignment for !0;~< has been running for at least 1031221ms
	java.lang.Exception: Assignment of !0;~<
		at sun.misc.Unsafe.park(Native Method)
		at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
		at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836)
		at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireQueued(AbstractQueuedSynchronizer.java:870)
		at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:1199)
		at java.util.concurrent.locks.ReentrantLock$NonfairSync.lock(ReentrantLock.java:209)
		at java.util.concurrent.locks.ReentrantLock.lock(ReentrantLock.java:285)
		at org.apache.accumulo.tserver.tablet.Tablet.removeInUseLogs(Tablet.java:2465)
		at org.apache.accumulo.tserver.TabletServer$12.removeInUse(TabletServer.java:3413)
		at org.apache.accumulo.tserver.TabletServer.findOldestUnreferencedWals(TabletServer.java:3370)
		at org.apache.accumulo.tserver.TabletServer.markUnusedWALs(TabletServer.java:3421)
		at org.apache.accumulo.tserver.TabletServer.minorCompactionFinished(TabletServer.java:3245)
		at org.apache.accumulo.tserver.tablet.DatafileManager.bringMinorCompactionOnline(DatafileManager.java:459)
		at org.apache.accumulo.tserver.tablet.Tablet.minorCompact(Tablet.java:914)
		at org.apache.accumulo.tserver.tablet.MinorCompactionTask.run(MinorCompactionTask.java:90)
		at org.apache.accumulo.tserver.tablet.Tablet.minorCompactNow(Tablet.java:1047)
		at org.apache.accumulo.tserver.TabletServer$AssignmentHandler.run(TabletServer.java:2388)
		at org.apache.accumulo.fate.util.LoggingRunnable.run(LoggingRunnable.java:35)
		at org.apache.accumulo.tserver.ActiveAssignmentRunnable.run(ActiveAssignmentRunnable.java:64)
		at org.apache.htrace.wrappers.TraceRunnable.run(TraceRunnable.java:57)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
		at org.apache.accumulo.fate.util.LoggingRunnable.run(LoggingRunnable.java:35)
		at java.lang.Thread.run(Thread.java:748)
```

### PR DESCRIPTION
**Pull request by keith-turner**

### CODE REVIEW

**Comment by keith-turner** [comment]
I ran WriteAheadLogIT and looked at the debug logs to see if this change was working as expected.  Saw the following which showed the set operations were working as expected.

```
2018-06-29 16:39:25,397 [tablet.Tablet] DEBUG: Logs for memory compacted: 1<< 38577b31-28a0-46d8-9467-84d2952e1d17
2018-06-29 16:39:25,398 [tablet.Tablet] DEBUG: Logs for memory compacted: 1<< 697f2016-6e73-454b-b550-1d882272c520
2018-06-29 16:39:25,398 [tablet.Tablet] DEBUG: Logs for memory compacted: 1<< ad2640e2-001a-4835-b2a2-5bbcc9eeb374
2018-06-29 16:39:25,398 [tablet.Tablet] DEBUG: Logs for current memory: 1<< 697f2016-6e73-454b-b550-1d882272c520
2018-06-29 16:39:25,398 [tablet.Tablet] DEBUG: Logs to be destroyed: 1<< 38577b31-28a0-46d8-9467-84d2952e1d17
2018-06-29 16:39:25,398 [tablet.Tablet] DEBUG: Logs to be destroyed: 1<< ad2640e2-001a-4835-b2a2-5bbcc9eeb374
```

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java [hunk 1]
```diff
@@ -2486,48 +2478,47 @@ public void removeInUseLogs(Set<DfsLogger> candidates) {
...
 
   synchronized void finishClearingUnusedLogs() {
-    removingLogs = false;
+    doomedLogs.clear();
     logLock.unlock();
   }
 
   private Set<DfsLogger> otherLogs = Collections.emptySet();
-  private boolean removingLogs = false;
+  private Set<DfsLogger> doomedLogs = new HashSet<>();
```
Comment (ctubbsii) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java]: A short comment to explain what "doomed" means in this context would be helpful.

**Comment by keith-turner** [comment]
All ITs passed when run from b03b094

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java [hunk 2]
```diff
@@ -2518,6 +2518,7 @@ synchronized void finishClearingUnusedLogs() {
   }
 
   private Set<DfsLogger> otherLogs = Collections.emptySet();
+  // logs related to minor compaction file being added to the metadata table
```
Comment (ctubbsii) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java]: This wording is a bit confusing. Can you expand on how they are related? Something like "Doomed logs are logs leftover from task X and will be deleted" (or whatever).

**Pull request merged**